### PR TITLE
fixed image save for magic cam, added save support for midjourney

### PR DESCRIPTION
**Pull request by pnivek**
- Rabbit's latest cloud update added freestyle magic cams which changed the API responses for getting files. This resulted in the saving feature breaking for magic cams. This change includes a refactor that fixes this issue and moves the save resources method to the journal class instead of the entry one.
- option now available to also save the midjourney preview images that appear in rabbit hole entries (256x256).
- also updated save function in LAH self int to properly handle different types of paths. the current version was leading to unexpected behavior. now it functions the way the user would expect given mac/window style paths
- also made some of the debug logging better when printing entries
- tested using different path scenarios and saw it work

### CODE REVIEW

File: .gitignore
Comment (dot-Justin) [inline .gitignore:1]: Should we be pushing changes to this file?

Comment (pnivek) [reply .gitignore:1]: probably not. only files common to everyone should be and this tests folder isn't. with that being said should we clean up this file then?

Comment (pnivek) [reply .gitignore:1]: resolved.

**Review by dot-Justin — APPROVED** [review]
LGTM!

**Pull request merged**